src/test/java/uk/gov/pay/adminusers/infra/SqsTestDocker.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.pay.adminusers.infra;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class SqsTestDocker {
+    private static final Logger logger = LoggerFactory.getLogger(SqsTestDocker.class);
+
+    private static GenericContainer sqsContainer;
+
+    public static AmazonSQS initialise(String... queues) {
+        try {
+            createContainer();
+            return createQueues(queues);
+        } catch (Exception e) {
+            logger.error("Exception initialising SQS Container - {}", e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void createContainer() {
+        if (sqsContainer == null) {
+            logger.info("Creating SQS Container");
+
+            sqsContainer = new GenericContainer("mvisonneau/alpine-sqs:1.2.0")
+                    .withExposedPorts(9324)
+                    .waitingFor(Wait.forHttp("/?Action=GetQueueUrl&QueueName=default"));
+
+            sqsContainer.start();
+        }
+    }
+
+    public static void stopContainer() {
+        sqsContainer.stop();
+        sqsContainer = null;
+    }
+
+    private static AmazonSQS createQueues(String... queues) {
```

Comment:
Have you tried saving doing a null check by passing in a `List<String> queues` and then doing `queues.forEach(q -> amazonSQS.createQueue(q))`? Code would look cleaner too.

Reply:
Yeh happy to make this change, this is another bit I copied from ledger without reading the code that much.